log4j2-logstash-webflux-example/src/main/java/se/fnord/logtags/log4j2_logstash/examples/webflux/Main.java

```diff
@@ -85,6 +94,9 @@ public Mono<ServerResponse> logTagsFunction(ServerRequest request) {
         .doOnEach(log.infoTags(s -> Tags.of(
             "test_status", s.statusCode().value(),
             "message", "Have a look: " + s.statusCode().value())))
+        .doOnEach(LOG.onNext(INFO, t -> t
+            .add("test_status", s -> s.statusCode().value())
+            .message(s -> "Have a look: " + s.statusCode().value())))
```

Comment:
Interface design: several different tag might need access to the next value or throwable, do you think it would be much simpler to have the tags builder "value agnostic", like using the following format:
```
LOG.onNext(INFO, (builder, next) -> builder
            .add("test_status", next.statusCode().value())
            .message("Have a look: " + next.statusCode().value()))
```

This eliminates need of using a separate lambda for every tag, also saves a bit of memory.

Reply:
Hmm, I think I tried that already, and it didn't work out for some reason, but maybe I missed something!

I'll give it another go!

Reply:
Although those log bits are likely to be reworked at some point.

The stateful ones (counters, empty etc) in favour of the tap logger, the non-stateful ones could be much simpler and maybe even called through `doOnNext` instead of `doOnEach` to indicate they only deal with success values 

I just haven't built the necessary plumbing to do stateful tap loggers yet